Plugins/NativeCapture/Source/NativeCapture.cpp

```diff
@@ -0,0 +1,81 @@
+#include "NativeCapture.h"
+
+#include <Babylon/JsRuntime.h>
+#include <GraphicsImpl.h>
+
+#include <vector>
+
+namespace Babylon::Plugins::Internal
+{
+    class NativeCapture : public Napi::ObjectWrap<NativeCapture>
+    {
+    public:
+        static constexpr auto JS_CLASS_NAME = "NativeCapture";
+
+        static void Initialize(Napi::Env env)
+        {
+            Napi::HandleScope scope{env};
+
+            Napi::Function func = NativeCapture::DefineClass(
+                env,
+                JS_CLASS_NAME,
+                {
+                    NativeCapture::InstanceMethod("addOnCaptureCallback", &NativeCapture::AddOnCaptureCallback),
+                    NativeCapture::InstanceMethod("dispose", &NativeCapture::Dispose),
+                });
+
+            env.Global().Set(JS_CLASS_NAME, func);
+        }
+
+        NativeCapture(const Napi::CallbackInfo& info)
+            : Napi::ObjectWrap<NativeCapture>{info}
+            , m_runtime{JsRuntime::GetFromJavaScript(info.Env())}
+            , m_graphicsImpl(Graphics::Impl::GetFromJavaScript(info.Env()))
+            , m_ticket{m_graphicsImpl.AddCaptureCallback([this](auto& data) { CaptureDataReceived(data); })}
+        {
+            m_graphicsImpl.StartCapture();
+        }
+
+    private:
+        void AddOnCaptureCallback(const Napi::CallbackInfo& info)
+        {
+            auto listener = info[0].As<Napi::Function>();
+            m_callbacks.push_back(Napi::Persistent(listener));
+        }
+
+        void CaptureDataReceived(const BgfxCallback::CaptureData& data)
+        {
+            std::vector<uint8_t> bytes{};
```

Comment:
Related to my comment below, but rather than allocating a vector each frame, can we just allocate a JS `Uint8Array` early on that we hold onto for the lifetime of this object, and just copy the source data into it each frame?

Reply:
Maybe only recreate it if/when the data size changes, which in practice will probably be infrequent.

Reply:
Spoke offline. Unfortunately because the callback here is dispatched to the JS thread (starting from the graphics thread), we need to copy it to a memory location that will be long-lived enough to make it onto the JS thread even if the JS runtime is a bit behind; and to avoid accidentally paving over the last frame's data in case we receive two graphics thread callbacks before processing one on the JS side, we cannot assume it's safe to reuse the same buffer. If we want to avoid allocating these vectors every single frame, we could probably make a resource pool to create and recycle them as needed. That feels like a performance optimization, though, so it might be better to tackle that in a separate PR improving on this one once we start using it.